src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -260,7 +260,12 @@ def require_package
       begin
         @package = Package.get_by_project_and_name(@project.to_param, params[:package],
                                                    follow_project_links: true, follow_multibuild: true)
-      rescue APIException # why it's not found is of no concern :)
+      rescue APIException => exception # why it's not found is of no concern :)
+        flash[:error] = if exception.is_a?(Package::ReadSourceAccessError) || exception.is_a?(Package::ReadAccessError)
+                          "You don't have access to the package \"#{params[:package]}\""
+                        else
+                          "Package \"#{params[:package]}\" not found in project \"#{params[:project]}\""
```

Comment:
are we sure that it was not found? couldn't be a different ''APIException"?

Reply:
We kept the same message as before, but we moved it up here so whenever it's not one of the access errors, we simply use it.